src/main/scala/it/pagopa/pdnd/interop/uservice/partymanagement/api/impl/PartyApiServiceImpl.scala

```diff
@@ -236,25 +236,28 @@ class PartyApiServiceImpl(
 
   }
 
+  private def parseArrayParameter(param: String): List[String] =
+    if (param == "[]") List.empty
+    else param.parseCommaSeparated
+
   /** Code: 200, Message: successful operation, DataType: Relationships
     * Code: 400, Message: Invalid ID supplied, DataType: Problem
     */
   override def getRelationships(
     from: Option[String],
     to: Option[String],
-    role: Option[String],
-    state: Option[String],
-    product: Option[String],
-    productRole: Option[String]
+    roles: String,
+    states: String,
+    products: String,
+    productRoles: String
```

Comment:
these are still optional, but the generator is not able to handle arrays correctly (I hate it).
I preferred a better detailed interface specification over nicer code.
Let me know what you think

Reply:
Working on the template could help us in this direction?

Reply:
Unfortunately no, if I got it right the data type is generated [here](https://github.com/OpenAPITools/openapi-generator/blob/master/modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ScalaAkkaHttpServerCodegen.java#L277).
We could fork/make a PR, but it will take more time

Reply:
Funny note: the array implementation in the library for akka is really bad.
For array types (i.e. of Strings) it will always default them to `"new ListBuffer[String]()"`, and yes, I mean that exact string, not the result of that expression 😵‍💫

Reply:
what about having the query parameter as a simple string instead, so leaving to the client the ownership of sending a "comma separated sequence of xxx"? Probably this could require a validation layer at our side over a miswritten comma separated sequence, though.

Reply:
I was thinking that since the openapi specifications consider this specific case, it could be useful to leverage on it.
Think also on (frontend) frameworks that generate code starting from the openapi specs, this would render also their code more solid